Add retry mechanism, timezone handling, and JSON cleanup features

X-Abhishek-X opened the pull request:
This PR adds several improvements to make the downloader more robust and user-friendly:

Changes to main.py:
- Added automatic retry mechanism with exponential backoff (default 3 attempts)
- Implemented automatic timezone conversion - files now use system local time instead of UTC
- Added JSON cleanup feature that removes successfully downloaded entries from memories_history.json
- Creates backup before modifying JSON file
- Added --max-retries flag to configure retry attempts
- Added --no-cleanup flag to preserve original JSON file
- Improved error handling for failed downloads
- Enhanced progress tracking with download/skip/fail statistics
- Added graceful cancellation support with Ctrl+C

Changes to README.md:
- Rewrote documentation to be more concise and professional
- Streamlined installation and usage instructions
- Added troubleshooting section
- Clarified folder structure requirements

These changes make it easier to handle large memory exports, especially when dealing with network issues or interrupted downloads. The retry mechanism and JSON cleanup allow users to resume downloads without re-downloading existing files.

X-Abhishek-X committed:
Initial commit: Snapchat Memories bulk downloader with EXIF metadata

binnichtaktiv commented on README.md:
```diff
@@ -1,50 +1,152 @@
+1. Extract the ZIP file you received from Snapchat
+2. Locate `memories_history.json` (usually in a `json` subfolder)
+3. Create a `json` folder in your project directory
+4. Copy `memories_history.json` into the `json` folder
```
just wondering why do you do it like this if you have to give the path `json/memories_history.json` as arg later anyway? If you ask the user to create and copy it into the json folder then it would be nice to automatically detect the memories_history.json file in that json folder

X-Abhishek-X replied:
Hey @binnichtaktiv! Thanks for catching that! 

You're absolutely right - it is a bit redundant to tell users to put the file in the folder and then still make them type the path. That's a great point!

I'll update it so the script automatically looks for [json/memories_history.json] by default, and users only need to specify a path if they want to use a different location.

Great suggestion - I'll implement this!

binnichtaktiv commented on main.py:
```diff
@@ -33,9 +36,16 @@ def model_post_init(self, __context):
     @property
     def filename(self) -> str:
-        return self.date.strftime("%Y-%m-%d_%H-%M-%S")
+        """Generate filename using system's local timezone"""
+        return self.local_date.strftime("%Y-%m-%d_%H-%M-%S")
```
actually a good point but idk who will download snap memories at 3 am🤔

X-Abhishek-X replied:
> actually a good point but idk who will download snap memories at 3 am🤔

true lol

binnichtaktiv commented on README.md:
````diff
@@ -1,50 +1,174 @@
+python main.py /custom/path/memories_history.json -o ./my_memories
+```
+
+Re-download all files without EXIF metadata:
````
@X-Abhishek-X you can also fix this if you want. Copilot is actually useful for this... How many times can you do this per month for free?

X-Abhishek-X replied:
> @X-Abhishek-X you can also fix this if you want. Copilot is actually useful for this... How many times can you do this per month for free?

I don't have an issue with that I have GitHub pro student plan and other AI paid models as well for free.

X-Abhishek-X commented:
@binnichtaktiv How about now??

X-Abhishek-X commented:
Hi @ToTheMax, just wanted to give a friendly nudge on this PR since it's been open for a while and I'd love to get your feedback on it.

Since the initial submission I've addressed all the points raised in the review:
- Removed unused imports (signal, timedelta)
- Moved shutil import to the top level
- Timestamped backup filenames so previous backups are never overwritten
- Failed files are now printed in a summary at the end so users can see exactly what didn't download
- Fixed the unreachable return after the retry loop
- Fixed the inconsistent heading formatting in the README examples
- Auto-detects json/memories_history.json by default, as suggested by @binnichtaktiv

Happy to make any other adjustments you'd like before merging. Thanks for maintaining the project!

ToTheMax commented:
Hi @X-Abhishek-X thanks for your contribution!

However, something went wrong in this PR, all old commits are in the branch and there are merge conflicts, therefore I cannot easily review the changes you made

X-Abhishek-X committed:
feat: add retry mechanism, timezone handling, and JSON cleanup

X-Abhishek-X commented:
> Hi @X-Abhishek-X thanks for your contribution!
> 
> However, something went wrong in this PR, all old commits are in the branch and there are merge conflicts, therefore I cannot easily review the changes you made

Hi @ToTheMax , I've cleaned up the commit history and resolved all the merge conflicts. Squashed everything down to 1 clean commit. Should be ready for review!
